File: src/main/scala/WordCount.scala
object WordCount {
  def calculateWordCount(inputPath: String, outputPath: String, isLocal: Boolean): Unit = {
    /**
      * reference from https://spark.apache.org/examples.html
      */
    val spark = SparkUtils.getSparkSession(isLocal)
    val sc = spark.sparkContext
    val textFile = sc.textFile(inputPath)
    val counts = textFile.flatMap(line => line.split(" "))
      .map(word => (word, 1))
      .reduceByKey(_ + _)
    counts.saveAsTextFile(outputPath)
    sc.stop()
  }

  def main(args: Array[String]): Unit = {
    calculateWordCount(SparkUtils.pathPrefix + "const.txt", SparkUtils.pathPrefix + "wordcount_output", true)
  }
}

File: bigram-spark-python/wordcount.py
import pyspark
from pyspark import SparkContext
from pyspark import SparkConf
import sys

if __name__ == "__main__":
    conf=SparkConf().setAppName("wordcount").setMaster("local[*]")
    sc=SparkContext.getOrCreate(conf)

    text_file = sc.textFile(sys.argv[1])
    counts = text_file.flatMap(lambda line: line.split(" ")) \
                .map(lambda word: (word, 1)) \
                .reduceByKey(lambda a, b: a + b)
    counts.saveAsTextFile("files/wordcount")
    sc.stop()
